app/endpoints/nwb_validation.py

```diff
@@ -0,0 +1,365 @@
+    chunk_size = 1024 * 1024  # 1 MB
+    total_size = 0  # Track total size written
+
+    with tempfile.NamedTemporaryFile(delete=False, suffix=suffix) as temp_file:
+        temp_path = temp_file.name
+
+        try:
+            file.file.seek(0)  # Reset pointer
+            while True:
+                # Use chunk_size
+                chunk = file.file.read(chunk_size)
+                if not chunk:
+                    break
+
+                chunk_len = len(chunk)
+                total_size += chunk_len
+
+                # Check size limit before writing
+                if total_size > MAX_FILE_SIZE:
+                    _handle_file_too_large(temp_path)
+
+                temp_file.write(chunk)
+        except Exception:
+            if pathlib.Path(temp_path).exists():
+                pathlib.Path(temp_path).unlink(missing_ok=True)
+            raise
+        else:
+            return temp_path
+
+
+def _cleanup_temp_file(temp_path: str) -> None:
+    """Background task to clean up temporary file."""
+    if temp_path and pathlib.Path(temp_path).exists():
+        try:
+            pathlib.Path(temp_path).unlink()
+            L.debug(f"Cleaned up temp file: {temp_path}")
+        except OSError as e:
+            L.warning(f"Failed to delete temp NWB file: {e}")
+
+
+# Endpoint logic is now a standalone function
+def validate_nwb_file(
+    file: Annotated[UploadFile, File(description="NWB file to upload (.nwb)")],
+    background_tasks: BackgroundTasks,
+) -> NWBValidationResponse:
+    """Validates an uploaded .nwb file using registered readers."""
+    file_extension = pathlib.Path(file.filename).suffix.lower() if file.filename else ""
+    if file_extension != ".nwb":
+        raise HTTPException(
+            status_code=HTTPStatus.BAD_REQUEST,
+            detail={
+                "code": ApiErrorCode.BAD_REQUEST,
+                "detail": "Invalid file extension. Must be .nwb",
+            },
+        )
+
+    temp_file_path = ""
+
+    try:
+        # Save upload synchronously (still non-blocking for other requests)
+        temp_file_path = _save_upload_to_tempfile(file, suffix=".nwb")
+
+        if pathlib.Path(temp_file_path).stat().st_size == 0:
+            _handle_empty_file(file)
+
+        # Validate the file synchronously
+        validate_all_nwb_readers(temp_file_path)
+
+        # Schedule cleanup as a background task
+        background_tasks.add_task(_cleanup_temp_file, temp_file_path)
+
+        return NWBValidationResponse(
+            status="success",
+            message="NWB file validation successful.",
+        )
+
+    except FileTooLargeError:
+        L.error("NWB upload failed: File too large (Max: 50MB)")
+        # Cleanup is handled inside _save_upload_to_tempfile
+        raise HTTPException(
+            status_code=HTTPStatus.BAD_REQUEST,
+            detail={
+                "code": ApiErrorCode.BAD_REQUEST,
+                "detail": "Uploaded file is too large. Max size: 50 MB.",
```

Comment:
The string 50 MB is not up to date with MAX_FILE_SIZE and it should not be hardcoded. It should be enough to use the message contained in FileTooLargeError.